### DISCUSSION TITLE
dose pest support multi file ?

### DISCUSSION THREAD

**Discussion by zsytssk** [pest-parser/pest#629] [General]
dose pest support multi file?

**Comment by CAD97** [comment]
Can you clarify?

At the moment, a grammar must be specified in a single file.

**Comment by zsytssk** [comment]
@CAD97 
I want use pest write a lang;
split file make more organize and easy to maintain.

**Comment by baoyachi** [comment]
support example?

**Comment by zsytssk** [comment]
maybe something like this:
// basic_type.pest
```pest
number = @{
    "-"?
    ~ ("0" | ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*)
    ~ ("." ~ ASCII_DIGIT*)?
    ~ (^"e" ~ ("+" | "-")? ~ ASCII_DIGIT+)?
}

char = _{
    !("\"" | "\\") ~ ANY
    | "\\" ~ ("\"" | "\\" | "/" | "b" | "f" | "n" | "r" | "t")
    | "\\" ~ ("u" ~ ASCII_HEX_DIGIT{4})
}
inner = @{ char* }
string = ${ "\"" ~ inner ~ "\"" }

bool = {'false' | 'true'}
```

// value.pest
```pest
IMPORT {number, string, bool} from './basic_type';

value = {number | string | bool}
```

**Comment by sayrer** [comment]
The way I support this is to use a Bazel (or equivalent) build system that can track dependencies between files. Running `cat` on two files is a simple but compelling example.

https://docs.bazel.build/versions/master/be/general.html#genrule_examples

**Comment by thedodd** [comment]
It would definitely be quite nice to support a native import like system. Even if it was a simple native concat operation.

**Comment by singalen** [comment]
Modules are planned in #333 ("Modules" section), but the project desperately needs active maintainers.
